edb_stat_statements/edb_stat_statements.c

```diff
@@ -699,22 +696,12 @@ edbss_json_scalar(void *semstate, char *token, JsonTokenType tokenType) {
 			}
 			goto fail;
 
-		case EDB_STMT_INFO_PARSE_QUERY_ID:
-			if (tokenType == JSON_TOKEN_NUMBER) {
-				char *endptr;
-				uint64 query_id = strtoull(token, &endptr, 10);
-				if (*endptr == '\0' && query_id != UINT64_MAX) {
-					state->info->query_id = query_id;
-					pfree(token);
-					break;
-				}
-			}
-			goto fail;
-
-		case EDB_STMT_INFO_PARSE_CACHE_KEY:
+		case EDB_STMT_INFO_PARSE_ID:
 			if (tokenType == JSON_TOKEN_STRING) {
-				Datum cache_key = DirectFunctionCall1(uuid_in, CStringGetDatum(token));
-				state->info->cache_key = DatumGetUUIDP(cache_key);
+				Datum id_datum = DirectFunctionCall1(uuid_in, CStringGetDatum(token));
+				pg_uuid_t *id_ptr = DatumGetUUIDP(id_datum);
```

Comment:
Can these fail?

Reply:
Yes. The outer-most `PG_CATCH` in the current session will - like you said - do a longjmp and recover by sending an error to the peer. Such error will be propagated to the client as an `edb.errors.InvalidValueError`:

```
ERROR 116169 - 2024-11-20T11:22:27.377 postgres: invalid input syntax for type uuid: "b2f8e457-a4f8-ab73-1979-afb333f9c"
INFO 116169 - 2024-11-20T11:22:27.377 postgres: -- {"query": "select\n    (<__std__::int64>$0 + <__std__::int64>$1)", "type": 1, "extras": "{\"cc\": {\"__internal_no_apply_query_rewrites\": false, \"__internal_query_reflschema\": false, \"__internal_testmode\": false, \"allow_bare_ddl\": \"AlwaysAllow\", \"allow_dml_in_functions\": false, \"allow_user_specified_id\": false, \"apply_access_policies\": true, \"force_database_error\": \"false\", \"query_cache_mode\": \"Default\", \"simple_scoping\": null, \"store_migration_sdl\": \"NeverStore\", \"warn_old_scoping\": null}, \"pv\": [3, 0], \"of\": \"BINARY\", \"e1\": false, \"il\": 101, \"ii\": false, \"in\": true, \"io\": false, \"dn\": \"default\"}", "id": "b2f8e457-a4f8-ab73-1979-afb333f9c"}
INFO 116169 - 2024-11-20T11:22:27.377 postgres: 	SELECT edgedb_v6_2f20a50ab0.__qh_bd20a1eba9bb696335db87182e5b207f(($1)::int8, ($2)::int8)
---------------------------------------------------------------------- Exception occurred: invalid input syntax for type std::uuid: "b2f8e457-a4f8-ab73-1979-afb333f9c" ----------------------------------------------------------------------

1. edb.errors.InvalidValueError: invalid input syntax for type std::uuid: "b2f8e457-a4f8-ab73-1979-afb333f9c"

------------------------------------------------------------------------------------------------------------------ Details -------------------------------------------------------------------------------------------------------------------


edb.errors.InvalidValueError: invalid input syntax for type std::uuid: "b2f8e457-a4f8-ab73-1979-afb333f9c"
ERROR 116111 _localdev 2024-11-20T11:22:27.377 asyncio: an error in edgedb protocol
protocol: <edb.server.protocol.binary.EdgeConnection object at 0x73c6cfabf370>
transport: <uvloop.loop._SSLProtocolTransport object at 0x73c6d48086c0>
```